scipy/spatial/src/distance_impl.h

```diff
@@ -425,6 +426,78 @@ compute_mean_vector(double *res, const double *X, npy_intp num_rows, const npy_i
 }
 #endif
 
+#define DEFINE_CDIST(name, type) \
```

Comment:
No changed in `DEFINE_CDIST` and `DEFINE_PDIST` as far as I can tell, was there a reason to move them?

Reply:
The functions for `city_block`, `euclidean` and `chebyshev` need to be at least declared before `minkowski`. Instead of adding forward declarations, I just moved the definitions higher up.